### PR TITLE
Update dependent mysql version to 2.15.0

### PR DESCRIPTION
**Pull request by mk2**
Current dependent version (2.10.2) has lacked mysql_native_password auth switch request for Azure, so sails has not able to connect MySql on Microsoft Azure Cloud service.

Please update MySQL version for supporting auth-switch support (especially MySQL for Azure).

NOTE: https://github.com/mysqljs/mysql/blob/master/Changes.md#v2150-2017-10-05

### CODE REVIEW

File: test/queryable/parse-native-query-error.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ describe('Queryable ::', function() {
         assert.equal(err.exit, 'queryFailed');
 
         Pack.parseNativeQueryError({
-          nativeQueryError: err.output.error
+          nativeQueryError: err.raw.error
```
Comment (mikermcneil) [inline test/queryable/parse-native-query-error.js]: @mk2 would you mind sending this as a separate PR?

**Review by mikermcneil — CHANGES_REQUESTED** [review]

File: test/queryable/parse-native-query-error.js (hunk 1 above)
Comment (mk2) [reply test/queryable/parse-native-query-error.js]: @mikermcneil this was fixed in latest machinepack-mysql master branch, so I was pulled it and eliminated from this PR.

**Comment by mk2** [comment]
Hmm, all tests were passed in my local environment. In travis-ci environment, default mocha timeout (2000ms) may not be suffiecient. (I assumed travis-ci mysql was too slow.)

**Comment by xbassols** [comment]
Do you guys have a timeline on when this will be implemented? We are using MySQL Azure and need the dependency updated or otherwise sails-mysql does not work for us.

**Comment by mk2** [comment]
@xbassols
If you needed it right now, you may also upgrade it by yourself (see my https://github.com/mk2/machinepack-mysql-adhoc).

**Comment by mikermcneil** [comment]
@xbassols @mk2  Taking a look now

**Review by mikermcneil — APPROVED** [review]

**Comment by mikermcneil** [comment]
This closes https://github.com/balderdashy/sails/issues/4458

**Comment by mikermcneil** [comment]
@xbassols @mk2 @dougwilson  Thanks everyone!  Merged and publishing now.

**Pull request merged**